File: app/src/main/java/com/limayeneha/twitter_trov/ui/TweetsAdapter.java
package com.limayeneha.twitter_trov.ui;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.limayeneha.twitter_trov.R;

import com.limayeneha.twitter_trov.databinding.TweetsListViewBinding;
import com.limayeneha.twitter_trov.model.Tweet;
import com.limayeneha.twitter_trov.viewmodel.ItemTweetViewModel;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by limayeneha on 2/9/17.
 */

public class TweetsAdapter extends RecyclerView.Adapter<TweetsAdapter.PViewHolder> {

    private List<Tweet> tweets = new ArrayList<Tweet>();

    public class PViewHolder extends RecyclerView.ViewHolder {
        private TweetsListViewBinding mTweetsListViewBinding;

        public PViewHolder(TweetsListViewBinding tweetsListViewBinding) {
            super(tweetsListViewBinding.itemTweet);
            this.mTweetsListViewBinding = tweetsListViewBinding;
        }

        void bindTweet(Tweet tweet) {
            if (mTweetsListViewBinding.getTweetViewModel() == null) {
                mTweetsListViewBinding.setTweetViewModel(
                        new ItemTweetViewModel(tweet));
            } else {
                mTweetsListViewBinding.getTweetViewModel().setTweet(tweet);
            }
        }
    }

    public TweetsAdapter() {
    }

    public void setTweets(List<Tweet> newTweets) {
        tweets.clear();
        tweets.addAll(newTweets);
        notifyDataSetChanged();
    }

    @Override
    public PViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        TweetsListViewBinding itemTweetBinding =
                DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.tweets_list_view,
                        parent, false);
        return new PViewHolder(itemTweetBinding);
    }

    @Override
    public void onBindViewHolder(PViewHolder holder, int position) {
        holder.bindTweet(tweets.get(position));
    }

    @Override
    public int getItemCount() {
        return tweets.size();
    }


}

File: app/src/main/java/com/limayeneha/twitter_trov/ui/TweetsAdapter.kt
package com.limayeneha.twitter_trov.ui

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import com.limayeneha.twitter_trov.R

import com.limayeneha.twitter_trov.databinding.TweetsListViewBinding
import com.limayeneha.twitter_trov.model.Tweet
import com.limayeneha.twitter_trov.viewmodel.ItemTweetViewModel

import java.util.ArrayList

/**
 * Created by limayeneha on 2/9/17.
 */

class TweetsAdapter : RecyclerView.Adapter<TweetsAdapter.PViewHolder>() {

  private val tweets = ArrayList<Tweet>()

  inner class PViewHolder(private val mTweetsListViewBinding: TweetsListViewBinding) : RecyclerView.ViewHolder(
      mTweetsListViewBinding.itemTweet) {

    internal fun bindTweet(tweet: Tweet) {
      if (mTweetsListViewBinding.tweetViewModel == null) {
        mTweetsListViewBinding.tweetViewModel = ItemTweetViewModel(tweet)
      } else {
        mTweetsListViewBinding.tweetViewModel.setTweet(tweet)
      }
    }
  }

  fun setTweets(newTweets: List<Tweet>) {
    tweets.clear()
    tweets.addAll(newTweets)
    notifyDataSetChanged()
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PViewHolder {
    val itemTweetBinding = DataBindingUtil.inflate<TweetsListViewBinding>(LayoutInflater.from(parent.context),
        R.layout.tweets_list_view,
        parent, false)
    return PViewHolder(itemTweetBinding)
  }

  override fun onBindViewHolder(holder: PViewHolder, position: Int) {
    holder.bindTweet(tweets[position])
  }

  override fun getItemCount(): Int {
    return tweets.size
  }


}
